autofaiss/external/quantize.py

```diff
@@ -114,6 +114,10 @@ def build_index(
         Folder to save the temporary small indices that are generated by each spark executor.
         Only used when distributed = "pyspark".
     """
+    index_path = make_path_absolute(index_path)
```

Comment:
this can be done (in case we don't want to save the index), can you check it ?

Reply:
I just checked, it works when `save_on_disk` is False. But I think that we should make `index_path` as an optional argument explicitly. (the default value is kept, i.e. knn.index)
I will do another commit to:

- avoid using `make_path_absolute` if `index_path` is None
- raise an valueerror when `index_path` is None but `save_on_disk` is True

Reply:
Ok, I eventually print the error for the above 2nd case and return None as you have done for other edge cases.